framework/TSA/TimeSeriesAnalyzer.py

```diff
@@ -31,23 +31,25 @@ class TimeSeriesAnalyzer(utils.metaclass_insert(abc.ABCMeta, object)):
   ## define the clusterable features for this trainer.
   _features = []
 
-  def canGenerate(self) -> bool:
+  @classmethod
+  def canGenerate(cls) -> bool:
     """
     A predicate function to determine if object instance inherits from TimeSeriesGenerator.
 
     @ In, None
     @ Out, boolean, True if instance can generate
     """
-    return isinstance(self, TimeSeriesGenerator)
+    return issubclass(cls, TimeSeriesGenerator)
 
-  def canCharacterize(self) -> bool:
+  @classmethod
+  def canCharacterize(cls) -> bool:
     """
     A predicate function to determine if object instance inherits from TimeSeriesCharacterizer.
 
     @ In, None
     @ Out, boolean, True if instance can characterize
     """
-    return isinstance(self, TimeSeriesCharacterizer)
+    return issubclass(cls, TimeSeriesCharacterizer)
```

Comment:
Yeah, this came up because we can use it in the InputParams to pick the allowable classes rather than check this at run time. However, the InputParams gets constructed before any instances are instantiated, so checking the subclass instead made this more flexible. Let me know what you think.